kolibri/core/hooks.py

```diff
@@ -62,49 +63,52 @@ def plugin_url(self, plugin_class, url_name):
         return plugin_url(plugin_class, url_name)
 
     class Meta:
-
         abstract = True
 
 
-class MultipleThemesWarning(UserWarning):
-    pass
-
-
-class ThemeHook(KolibriHook):
+class ThemeHook(hooks.KolibriHook):
     """
     A hook to allow custom theming of Kolibri
+    Use this tool to help generate your brand colors: https://materialpalettes.com/
     """
 
-    class Meta:
-
-        abstract = True
+    def validateBrandColors(self, theme):
+        if "brandColors" not in theme:
+            logger.error("brand colors not defined by theme")
+            return False
+        required_colors = ["primary", "secondary"]
+        required_keys = [
+            "v_50",
+            "v_100",
+            "v_200",
+            "v_300",
+            "v_400",
+            "v_500",
+            "v_600",
+            "v_700",
+            "v_800",
+            "v_900",
+        ]
+        for color in required_colors:
+            if color not in theme["brandColors"]:
+                logger.error("'{}' not defined by theme".format(color))
+            for key in required_keys:
+                if key not in theme["brandColors"][color]:
+                    logger.error("{} '{}' not defined by theme".format(color, key))
 
     @property
-    @abstract_method
+    @hooks.only_one_registered
     def theme(self):
-        default = {
-            # Whether to show the Kolibri log
-            # Boolean
-            "showKolibriLogo": True,
-            # URL for custom logo
-            "customLogoURL": None,
-            # URL for custom login background image
-            "splashBackgroundURL": None,
-            # Color Palette specification
-            "paletteColors": {},
-            # Brand Color specification
-            "brandColors": {},
-            # Mapping from colors to particular usage
-            "tokenMapping": {},
-        }
-        theme = {}
-        once = False
-        for hook in self.registered_hooks:
-            if once:
-                warnings.warn("Multiple themes defined by plugins, ignoring all themes")
-                return default
-            for key in default:
-                theme[key] = getattr(hook, key, theme[key])
-            once = True
-
-        return theme or default
+        theme = self.registered_hooks[0].theme
+
+        self.validateBrandColors(theme)
+
+        # if a background image has been locally set using the `manage background` command, use it
+        if os.path.exists(os.path.join(settings.MEDIA_ROOT, "background.jpg")):
```

Comment:
Works for now - do we not allow pngs?

Might be good to store the filename somewhere instead.